evcouplings/complex/protocol.py

```diff
@@ -22,19 +22,68 @@
 )
 from evcouplings.align.protocol import modify_alignment
 
+from evcouplings.align.alignment import Alignment
+
 from evcouplings.complex.alignment import (
     write_concatenated_alignment
 )
 from evcouplings.complex.distance import (
     find_possible_partners, best_reciprocal_matching,
-    plot_distance_distribution
+    plot_distance_distribution, get_distance
 )
 from evcouplings.complex.similarity import (
     read_species_annotation_table,
     most_similar_by_organism,
     filter_best_reciprocal,
     find_paralogs
 )
+from evcouplings.couplings.mapping import (
+    SegmentIndexMapper, Segment, segment_map_positions_single_column
+)
+
+
+def _split_annotation_string(string):
+    id, annotation_str = string.split("/")
+    region_start, region_end = annotation_str.split("-")
+    return (id, int(region_start), int(region_end))
+
+
+def remove_overlapping_ids(id_dataframe):
+    """
+    Removes pairs of ids that are from the same sequence
+    and overlap by at least a certain amount
+
+    Paramters
+    ---------
+    id_dataframe: pd.DataFrame
+        contains columns id_1, id_2
+    amount_overlap_allowed: int
+        threshold of amount overlap allowed between two sequecnes
+
+    Returns
+    pd.Dataframe
+    """
+    temp_dataframe = id_dataframe.copy()
+
+    id_strings = [_split_annotation_string(x) for x in temp_dataframe.id_1]
+    temp_dataframe["id_string_1"], temp_dataframe["r_s_1"], temp_dataframe["r_e_1"] = \
```

Comment:
not a big fan of new lines with backslash - nicer to wrap first part of assignment in (...)

Reply:
I can't picture what you mean here, sorry, was only using the slash because the assignment doesn't seem to nicely support wrapping with ()